ArduCopter/system.cpp

```diff
@@ -140,11 +140,17 @@ void Copter::init_ardupilot()
     AP::compass().set_log_bit(MASK_LOG_COMPASS);
     AP::compass().init();
 
+    // initialise airspeed sensor
+    airspeed.init();
+
 #if OPTFLOW == ENABLED
     // make optflow available to AHRS
     ahrs.set_optflow(&optflow);
 #endif
 
+    // give AHRS the airspeed sensor
+    ahrs.set_airspeed(&airspeed); 
```

Comment:
AHRS should be using singleton.

Reply:
So this would affect plane as well, correct?

Reply:
@peterbarker it depends. Are there any code paths for which the EKF using airspeed would degrade anything? Copter isn't setting the fly forward assumption so I think it will be okay, but there are a lot of aspects associated that change if fly_forward was set which wouldn't be appropriate on copter. I'm just struggling to prove that as long as this isn't set it's safe.

Reply:
Wind state estimation and airspeed fusion are inhibited unless _ahrs->get_fly_forward() returns true.